src/core.jl

```diff
@@ -0,0 +1,70 @@
+function reproject(input_data, output_projection; shape_out = nothing, order = 1, hdu_in = 1, hdu_out = 1, return_footprint = true)
```

Comment:
I don't want the `return_footprint` argument, as it makes the function type unstable.  Remove it and always return the footprint (as explained below).  This is one of the Julia performance tips that I told you to keep in mind.  Please, read them again.

In general, always make sure that the function is type stable with `@code_warntype`.  I haven't run but I'm already sure that you didn't :wink:.